sdk/python/symbolchain/nem/SharedKey.py

```diff
@@ -0,0 +1,23 @@
+import sha3
+
+from ..CryptoTypes import SharedKey256
+from ..external import ed25519
+from ..SharedKey import SharedKey as BasicSharedKey
+
+
+class SharedKey(BasicSharedKey):
+	@staticmethod
+	def derive_shared_key(key_pair, other_public_key):
+		"""Derives shared encryption key from private key and public key."""
+
+		# note: key_pair.private_key is "unreversed", so need to reverse here again
+		return BasicSharedKey._derive_shared_key(other_public_key.bytes, key_pair.private_key.bytes[::-1], b'nem-nis1', sha3.keccak_512)
+
+	@staticmethod
+	def derive_shared_key_deprecated(key_pair, other_public_key, salt):
+		"""Derives shared encryption key from private key, public key and salt."""
```

Comment:
might want to document how this is different and/or at least mention it is deprecated